src/server/web_server.js

```diff
@@ -90,10 +91,14 @@ async function start_web_server() {
         const https_server = https.createServer({ ...ssl_cert, honorCipherOrder: true }, app);
         server_rpc.rpc.register_ws_transport(https_server);
         await P.ninvoke(https_server, 'listen', https_port);
+
     } catch (err) {
         dbg.error('Web Server FAILED TO START', err.stack || err);
         process.exit(1);
     }
+
+    // Try to start the metrics server.
+    prom_reporting.start_server(config.WS_METRICS_SERVER_PORT);
```

Comment:
missing await

Reply:
actually why not move it under the same try catch as above?

Reply:
Failing to report metrics, from a customer side, has different much lower severity then denial of service, so we agreed (RedHat's noobaa team) that it should not exist in that case. 
This is also why I didn't merge the code with the server code try and catch block. 

Now  `start_server` cannot fail because it will catch all rejects, log them, and then ignore. which means await is not needed as it is an "initiate and don't wait for" kind of operation.

Reply:
But the current code just leaves it unhandled, and even with retries this will eventually fail, and then what?

Reply:
We need better error handling here.

Reply:
@guymguym It is handled, `start_server` literally cannot fail (in the current implementation)

Reply:
still, when you don't await an async function it just feels wrong.
both for the possible new error conditions we might introduce inside start_server as time goes by,
and also because we might need to add more setup code and making it ordered is simpler.

Reply:
Done, I am now waiting for all of them, just for the sake of consistency